internal/provider/data_source_projects_test.go

```diff
@@ -0,0 +1,133 @@
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_projects.all", "projects.0.id"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.name", "Default Project"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.description", ""),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.organization", orgName),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_projects.all", "projects.1.id"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.name", "project1"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.description", "Project 1"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.1.organization", orgName),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_projects.all", "projects.2.id"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.name", "project2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.description", "Project 2"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.2.organization", orgName),
+					resource.TestCheckResourceAttrSet(
+						"data.tfe_projects.all", "projects.3.id"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.name", "project3"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.description", "Project 3"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.3.organization", orgName),
+				),
+			},
+		},
+	})
+}
+
+func TestAccTFEProjectsDataSource_basicNoProjects(t *testing.T) {
+	rInt := rand.New(rand.NewSource(time.Now().UnixNano())).Int()
+	orgName := fmt.Sprintf("tst-terraform-%d", rInt)
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV5ProviderFactories: testAccMuxedProviders,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccTFEProjectsDataSourceConfig_noProjects(orgName),
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "organization", orgName),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.#", "1"),
+					resource.TestCheckResourceAttr(
+						"data.tfe_projects.all", "projects.0.name", "Default Project"),
+				),
+			},
+		},
+	})
+}
+
+func testAccTFEProjectsDataSourceConfig(orgName string) string {
+	return fmt.Sprintf(`
+resource "tfe_organization" "organization" {
+  name  = "%s"
+  email = "admin@tfe.local"
+}
+
+resource "tfe_project" "project1" {
+  name         = "project1"
+  description  = "Project 1"
+  organization = tfe_organization.organization.name
+}
+
+resource "tfe_project" "project2" {
+  name        = "project2"
+  description = "Project 2"
+  organization = tfe_organization.organization.name
```

Comment:
I tried running the CI tests in a duplicate PR since it doesn't seem to run here, and I do see a failing test. 
```
data_source_projects_test.go:19: Step 1/1 error: Check failed: Check 2/18 error: data.tfe_projects.all: Attribute 'projects.#' expected "4", got "1"
        Check 7/18 error: data.tfe_projects.all: Attribute 'projects.1.id' expected to be set
        Check 8/18 error: data.tfe_projects.all: Attribute 'projects.1.name' not found
        Check 9/18 error: data.tfe_projects.all: Attribute 'projects.1.description' not found
        Check 10/18 error: data.tfe_projects.all: Attribute 'projects.1.organization' not found
        Check 11/18 error: data.tfe_projects.all: Attribute 'projects.2.id' expected to be set
        Check 12/18 error: data.tfe_projects.all: Attribute 'projects.2.name' not found
        Check 13/18 error: data.tfe_projects.all: Attribute 'projects.2.description' not found
        Check 14/18 error: data.tfe_projects.all: Attribute 'projects.2.organization' not found
        Check 15/18 error: data.tfe_projects.all: Attribute 'projects.3.id' expected to be set
        Check 16/18 error: data.tfe_projects.all: Attribute 'projects.3.name' not found
        Check 17/18 error: data.tfe_projects.all: Attribute 'projects.3.description' not found
        Check 18/18 error: data.tfe_projects.all: Attribute 'projects.3.organization' not found
```
https://github.com/hashicorp/terraform-provider-tfe/pull/1506